CMSgov/design-system
# WNMGDS-3037 Adds script to compute percentage of nightly tests that pass

## Summary

This script analyzes jenkins nightly build data to compute a general pass rate and outputs a list of failed builds for manual inspection.
**Note:** It utilizes the Puppeteer library to simulate logging into Jenkins as a workaround, pending resolution of issues with setting up a proper API token.

```
// Sample output for a range of builds
% npx ts-node get-nightly-build-pass-rate.ts 750 3
Logging in to Jenkins...
Login successful.
Analyzing builds from 748 to 750...
Fetching data for build #748...
Fetching data for build #749...
Fetching data for build #750...
Analysis complete.
Pass Count: 3
Fail Count: 0
Pass-to-Fail Rate: 100.00%
Failed Builds: 
```
[Jira ticket](https://jira.cms.gov/browse/WNMGDS-3037) 

## How to test
- Pull down the branch
- Ensure you are connected to the Zscaler VPN.
- Navigate to scripts directory:`cd scripts`
- Create a .env file in the scripts directory with the following variables:
```
JENKINS_ID=<your-eua-id>
JENKINS_PASSWORD=<your-password>
JENKINS_LOGIN_URL=<Jenkins login URL>
JENKINS_BASE_API_URL=<Jenkins base API URL>
```
- Run `yarn install`
- Run `npx ts-node get-nightly-build-pass-rate.ts <build number> <numberOfDays>`
  - Replace <build number> with the latest jenkins build number you want to inspect.
  - Replace <numberOfDays> with the range of days to analyze (optional; defaults to 1 if omitted).


## Checklist
